core/vm/environment.go

```diff
@@ -35,6 +37,7 @@ type (
 
 // Context provides the EVM with auxilary information. Once provided it shouldn't be modified.
 type Context struct {
+	context.Context
```

Comment:
Thinking about it more, it feels like we are running in circles here. The Interpreter contained in EVM is supposed to be reusable for some reason, so all runtime state (including depth, etc) must be held in locals or in the EVM struct. Now you say that you're not happy with EVM being single use. Maybe we should make Interpreter single use and EVM reusable/thread safe?